src/agent/platform.rs

```diff
@@ -136,172 +141,133 @@ impl Client {
 
 #[derive(Clone)]
 struct AuthToken {
-    token: Arc<Mutex<AsciiMetadataValue>>,
+    inner: Arc<Mutex<AsciiMetadataValue>>,
 }
 
 impl AuthToken {
-    fn new(val: &str) -> Result<Self> {
-        let token = format_bearer(val)?;
+    fn new(token: &str) -> Self {
+        let bearer = format_bearer(token);
 
-        Ok(Self {
-            token: Arc::new(Mutex::new(token)),
-        })
+        Self {
+            inner: Arc::new(Mutex::new(bearer)),
+        }
     }
 
-    fn get(&self) -> AsciiMetadataValue {
-        self.token.lock()
-            .unwrap()
-            .clone()
+    fn bearer(&self) -> AsciiMetadataValue {
+        self.inner.lock().unwrap().clone()
```

Comment:
I think it makes more sense for this method to format the inner attribute into a bearer token, rather than every time it gets set.

Reply:
conceptually yes but trying to save a few cycles.